src/components/GraphFilter/GraphFilter.tsx

```diff
@@ -1,30 +1,49 @@
 import * as React from 'react';
+import { observer } from 'mobx-react';
 import { ButtonGroup, DropdownButton, MenuItem, Toolbar } from 'patternfly-react';
 import { ButtonToolbar } from 'react-bootstrap';
 
 import { GraphFilterProps, GraphFilterState } from '../../types/GraphFilter';
-import * as API from '../../services/Api';
+import Namespace from '../../types/Namespace';
 import { DurationButtonGroup } from './DurationButtonGroup';
 import { LayoutButtonGroup } from './LayoutButtonGroup';
+import { namespaceStore } from '../../model/Namespace';
 
+type DropdownListType = {
+  list: Namespace[];
+  current: Namespace;
+  onSelect: Function;
+};
+
+export class DropdownList extends React.PureComponent<DropdownListType, {}> {
```

Comment:
I'm just reading about that ( https://reactjs.org/docs/react-api.html#reactpurecomponent )
So IIRC we should use PureComponent as much as possible, as I think many of our components rendering functions are only state/props dependant.